### PR TITLE
Updated bonded-finance docs

### PR DESCRIPTION
**Pull request by PoisonPhang**
Expanded the readme and clarified the language used in the source docs.

### CODE REVIEW

**Review by hussein-aitlahcen — DISMISSED** [review]

File: frame/bonded-finance/README.md [hunk 1]
```diff
@@ -1,3 +1,68 @@
-# Overview
+# Bonded Finance
+
+A pallet providing means of submitting and maintaining bond offers.
+
+## Bond Offer Workflow
```
Comment (dzmitry-lahoda) [inline frame/bonded-finance/README.md:5]: should we have docs in rs and in md? I think right is one place. i do not care what.
also, I should that explanatory should go first - what is value of pallet and what is mechanic. example is good, but if you can do summary - ok. i do not think that details should go first. high level introductory should guide user reading later docs.

Comment (PoisonPhang) [reply frame/bonded-finance/README.md:34]: As for your first suggestion, let me know what you think of these new changes.
As for the second, while I do see where you're coming from, I think there is value to having the readme and top-level cargo docs mirror each other in the case of pallets.

**Review by hussein-aitlahcen — DISMISSED** [review]

**Review by hussein-aitlahcen — DISMISSED** [review]

**Comment by dzmitry-lahoda** [comment]
so we copy docs into readme.md and into lib.rs?

**Review by dzmitry-lahoda — APPROVED** [review]

**Comment by PoisonPhang** [comment]
Rebased to implement newer documentation and amended README.md, please re-review

**Review by hussein-aitlahcen — DISMISSED** [review]

File: frame/bonded-finance/README.md [hunk 2]
```diff
@@ -1,3 +1,100 @@
...
+A pallet providing means of submitting and maintaining bond offers.
+
+## Overview
+
+The bonded finance pallet enables users to create bond offers, buy bonds from 
+other users, and cancel existing bond offers via admin intervention. These 
+bonds can be listed with various assets and can offer a different asset as the 
+reward.
+
+One bound offer may contain multiple identically priced bounds. Each bound will
```
Comment (benluelo) [inline frame/bonded-finance/README.md:12]: bound or bond?

File: frame/bonded-finance/README.md [hunk 3]
```diff
@@ -1,3 +1,100 @@
...
+
+Bond offers can be canceled with the `cancel` extrinsic. This can only be 
+successfully called by the `AdminOrigin`.
+
+Once canceled, the stake and liquidity will be returned to the offer creator. 
+However, this will not cancel currently vested rewards.
+
+## Technical Notes
+
+* This pallet implements the `composable_traits::bonded_finance::BondedFinance`.
```
Comment (benluelo) [inline frame/bonded-finance/README.md:99]: implements the [...] trait

File: frame/bonded-finance/src/lib.rs [hunk 4]
```diff
@@ -30,26 +30,22 @@
 	unused_extern_crates
 )]
 
-//! Bonded Finance pallet
+//! # Bonded Finance
```
Comment (benluelo) [inline frame/bonded-finance/src/lib.rs:33]: why not just use `#![doc = include_str!("../README.md")]`?

**Review by benluelo — CHANGES_REQUESTED** [review]

File: frame/bonded-finance/README.md (hunk 2 above)
Comment (PoisonPhang) [reply frame/bonded-finance/README.md]: ad593807111cc66cffec3fe50cf921ebedc21179

File: frame/bonded-finance/README.md (hunk 3 above)
Comment (PoisonPhang) [reply frame/bonded-finance/README.md]: ff16c9a96662adeffcae3b528d3098240aa7e5ea

File: frame/bonded-finance/src/lib.rs (hunk 4 above)
Comment (PoisonPhang) [reply frame/bonded-finance/src/lib.rs]: Didn't realize this was possible, thanks. 

09c51a49f669938a6716b531fa2e2f71109747af

**Review by benluelo — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**